packages/cardano-graphql-services/src/DataProjection/blockHandlers/helpers.ts

```diff
@@ -0,0 +1,79 @@
+import { Asset, BigIntMath } from '@cardano-sdk/core';
+import { AssetMetadata } from '../../MetadataClient/types';
+import { Schema, isAlonzoBlock, isMaryBlock } from '@cardano-ogmios/client';
+
+export const getBlockType = (block: Schema.Block) => {
+  let b: Schema.BlockMary | Schema.BlockAlonzo | undefined;
+  if (isAlonzoBlock(block)) {
+    b = block.alonzo as Schema.BlockAlonzo;
+  } else if (isMaryBlock(block)) {
+    b = block.mary as Schema.BlockMary;
+  }
+  return b;
+};
+
+export const mapMetadata = (serverAssetMetadata: AssetMetadata): Asset.TokenMetadata => ({
+  decimals: serverAssetMetadata.decimals.value,
+  desc: serverAssetMetadata.description?.value,
+  icon: serverAssetMetadata.logo?.value,
+  name: serverAssetMetadata.name?.value,
+  ticker: serverAssetMetadata.ticker?.value,
+  url: serverAssetMetadata.url?.value
+  // Missing fields: sizedIcons
+});
+
+const mapAssets = (assets: { [k: string]: Schema.AssetQuantity } | undefined) =>
+  // Missing: transactionOutput, asset
+  assets ? Object.entries(assets).map(([name, quantity]) => ({ asset: { name }, quantity })) : [];
+
+const mapOutput = (output: Schema.TxOut) =>
+  // Missing: index, transaction, datumHash
+  ({
+    address: output.address,
+    value: { assets: mapAssets(output.value.assets), coin: output.value.coins }
+  });
+
+const mapInput = (input: Schema.TxIn) =>
+  // Missing: address:, redeemer, sourceTransaction, transaction, value
+  ({ index: input.index });
+
+const mapAuxiliaryData = (metadata: Schema.AuxiliaryData) => ({
```

Comment:
This is a complex nested object (see [Metadatum](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/cardano-graphql-services/src/Schema/types/Transaction/AuxiliaryData/Metadatum.ts)). Let's leave it out of scope of this PR.